packages/dsm_address_project/src/hooks/useDebounce.ts

```diff
@@ -0,0 +1,16 @@
+import { useEffect, useState } from "react";
+
+export default function useDebounce(value: string, delay: number) {
+  const [debounceValue, setDebounceValue] = useState<string>(value);
+
+  useEffect(() => {
+    const timmer = setTimeout(() => {
+      setDebounceValue(value);
+    }, delay);
+    return () => {
+      clearTimeout(timmer);
+    };
+  }, [value]);
```

Comment:
debounce 로직이 이상한거같은데.. 
이렇게 되면 value가 변경되는 족족 다 setTimeout으로 설정한 액션들이 실행될거라, debounce를 건 이유가 사라질거에요.

만약 이렇게 한다면 다음과같이 바꿔야합니다.

```suggestion
   const timer = useRef<number>();
  useEffect(() => {
    window.clearTimeout(timer.current);
    timer.current = window.setTimeout(() => {
      setDebounceValue(value);
    }, delay);
    return () => {
      window.clearTimeout(timer.current);
    };
  }, [value]);
```

timer를 useEffect 바깥으로 빼내고, useRef를 활용해야합니다.
useEffect가 실행될 때, timer에 할당된 timeout 액션을 취소시키고 새로운 timeout를 할당시켜야 debounce 생각하는 debounce 로직이 구현될거에요

Reply:
기존 코드에서 return 안에서 clear를 했는데도 setTimeout이 실행되는 건가요?